src/data_source/storage/sql.rs

```diff
@@ -2211,4 +2821,314 @@ mod test {
+                        sql_param(&"randomString"),
+                        sql_param(&test_data),
+                    ],
+                )
+                .await
+                .unwrap();
+        // proof for the index from the tree
+        let (_, proof) = test_tree.lookup(0).expect_ok().unwrap();
+        // traversal path for the index.
+        let traversal_path =
+            <usize as ToTraversalPath<U8>>::to_traversal_path(&0, test_tree.height());
+        // insert merkle nodes
+        <SqlStorage as UpdateStateData<_, MockMerkleTree>>::insert_merkle_nodes(
+            &mut storage,
+            &test_tree,
+            proof.proof.clone(),
+            traversal_path.clone(),
+            block_height as u64,
+        )
+        .await
+        .expect("failed to insert nodes");
+        // the path from the db and and tree should match
+        let merkle_path = <SqlStorage as MerklizedStateDataSource<_, MockMerkleTree>>::get_path(
+            &storage,
+            &test_tree,
+            Snapshot::Index(block_height as u64),
+            0,
+        )
+        .await
+        .unwrap();
+
+        let (_, proof) = test_tree.lookup(0).expect_ok().unwrap();
+
+        tracing::info!("merkle path {:?}", merkle_path);
+
+        // merkle path from the storage should match the path from test tree
+        assert_eq!(merkle_path, proof.proof.clone(), "merkle paths mismatch");
+
+        //Deleting the index 0
+        test_tree.remove(0).expect("failed to delete index 0 ");
+
+        // Update the database with the proof
+        // Created = 2 in this case
+        let proof = test_tree.universal_lookup(0).expect_not_found().unwrap();
+
+        <SqlStorage as UpdateStateData<_, MockMerkleTree>>::insert_merkle_nodes(
+            &mut storage,
+            &test_tree,
+            proof.proof.clone(),
+            traversal_path.clone(),
+            2_u64,
+        )
+        .await
+        .expect("failed to insert nodes");
+        // Insert the new header
+        storage
+        .query_opt(
+            "INSERT INTO HEADER VALUES ($1, $2, 't', 0, $3) ON CONFLICT(height) DO UPDATE set data = excluded.data",
+            [
+                sql_param(&2_i64),
+                sql_param(&"randomString2"),
+                sql_param(&serde_json::json!({ test_tree.header_state_commitment_field() : serde_json::to_string(&test_tree.commitment()).unwrap()})),
+            ],
+        )
+        .await
+        .unwrap();
+        // Get non membership proof
+        let non_membership_path =
+            <SqlStorage as MerklizedStateDataSource<_, MockMerkleTree>>::get_path(
+                &storage,
+                &test_tree,
+                Snapshot::Index(2_u64),
+                0,
+            )
+            .await
+            .unwrap();
+        // Assert that the paths from the db and the tree are equal
+        assert_eq!(non_membership_path, proof.proof, "paths dont match");
+    }
```

Comment:
Lookup up again with created = 1 and ensure we can still retrieve the membership proof from that block height